### PR TITLE
Get jobs backend

### PR DESCRIPTION
**Pull request by vtruo009**
* Leaving the filtering as it is right now
* None of the fields are required because when user enters nothing, the following happens:
1. title = title (which is empty)
2. type = [""]
3. minSalary = 0
4. hoursPerWeek = 0
5. startDate = new Date() which is the current date

* Problems:
1. if a job has type = ['grader', 'reader'] and user enters 'grader' as a filter in the query string, this job will not get returned . All jobs with type = ['grade'] will be returned.
2. title can only be filtered if it is perfect match

### CODE REVIEW

**Comment by johan1505** [comment]
Lol I merged a pr and I think that made this pr have conflicts now. Can you resolve them?

File: src/server/src/routes/job.ts [hunk 1]
```diff
@@ -9,6 +9,8 @@ import {
     getJobs,
 } from '@modules/job';
 import logger from '@shared/Logger';
+import { parse } from 'path';
```
Comment (johan1505) [inline src/server/src/routes/job.ts]: What are these imports, parse and start, for?

**Review by johan1505 — CHANGES_REQUESTED** [review]
Looks good. Left some comments. Also, 2 binary files called .DS_Store and src/,DS_Store were pushed, can you remove them?

File: src/server/src/routes/job.ts (hunk 1 above)
Comment (vtruo009) [reply src/server/src/routes/job.ts]: I'm not sure. It was there when I gitpulled

Comment (johan1505) [reply src/server/src/routes/job.ts]: I don't think we need it. Can you remove it?

**Comment by johan1505** [comment]
I pushed a commit that attempts to do filtering with querybuilder API from typeORM. I figured out how to do substring matching. I left comments on the code changed, please read them. Also, I think you can add pagination using the querybuilder. https://typeorm.io/#/select-query-builder. Look at the pagination section from that link. Let me know if you have questions

File: src/server/src/routes/job.ts (hunk 1 above)
Comment (vtruo009) [reply src/server/src/routes/job.ts]: removed

**Review by johan1505 — APPROVED** [review]

**Pull request merged**